lib/dock.rb

```diff
@@ -0,0 +1,32 @@
+class Dock
+  attr_reader :name, :max_rental_time, :currently_rented_boats,
+              :revenue, :total_hours_by_rental_type, :charges
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @currently_rented_boats = []
+    @revenue = 0
+    @total_hours_by_rental_type = Hash.new(0)
+    @rentals_by_ccn = Hash.new([])
```

Comment:
This hash is a bit confusing, it actually looks like it is holding credit_card_numbers by boat, rather than rentals by credit_card_number.